src/include/sof/lib_manager.h

```diff
@@ -0,0 +1,204 @@
+ *   |                       |                            | <------                  |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            | Copy Library Data        |         |
+ *   |                       |                            | -----------------------> |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            |   (IF AUTH_API_ENABLED)  |         |
+ *   |                       |                            |  Verify Manifest         |         |
+ *   |                       |                            | -------------------------|-------> |
+ *   |                       |                            |  results                 |         |
+ *   |                       |                            | <------------------------|-------- |
+ *   |                       |                            |   (IF AUTH_API_ENABLED)  |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            | Update Library           |         |
+ *   |                       |                            | descriptors table        |         |
+ *   |                       |                            | -------                  |         |
+ *   |                       |                            |        |                 |         |
+ *   |                       |                            | <------                  |         |
+ *   |                       |       return status        |                          |         |
+ *   |                       | <------------------------- |                          |         |
+ *   | Complete IPC request  |                            |                          |         |
+ *   | <-------------------  |                            |                          |         |
+ *
+ * Driver                   IPC4                       Library                     MEMORY    ENTITY
+ *                         Handler                     Manager                               VERIF.
+ */
+
+#ifndef __SOF_LIB_MANAGER_H__
+#define __SOF_LIB_MANAGER_H__
+
+#include <stdint.h>
+#include <rimage/cavs/cavs_ext_manifest.h>
+#include <rimage/sof/user/manifest.h>
+
+#define LIB_MANAGER_MAX_LIBS				16
+#define LIB_MANAGER_LIB_ID_SHIFT			12
+#define LIB_MANAGER_LIB_NOTIX_MAX_COUNT		4
+
+#define LIB_MANAGER_GET_LIB_ID(module_id) ((module_id) >> LIB_MANAGER_LIB_ID_SHIFT)
+#define LIB_MANAGER_GET_MODULE_INDEX(module_id) ((module_id) & 0xFFF)
+
+#ifdef CONFIG_LIBRARY_MANAGER
+struct ipc_lib_msg {
+	struct ipc_msg *msg;
+	struct list_item list;
+};
+
+static SHARED_DATA struct ext_library {
+	struct k_spinlock lock;	/* locking mechanism */
+	struct sof_man_fw_desc *desc[LIB_MANAGER_MAX_LIBS];
```

Comment:
better say here what data is protecting than `locking mechanism`. `locking mechanism` is too generic.